pennylane_lightning_kokkos/lightning_kokkos.py

```diff
@@ -245,12 +249,24 @@ def probability(self, wires=None, shot_range=None, bin_size=None):
     def expval(self, observable, shot_range=None, bin_size=None):
         if observable.name in [
             "Projector",
-            "Hamiltonian",
-            "SparseHamiltonian",
         ]:
             self.syncD2H()
             return super().expval(observable, shot_range=shot_range, bin_size=bin_size)
 
+        if observable.name in ["SparseHamiltonian"]:
+            CSR_SparseHamiltonian = observable.sparse_matrix().tocsr()
+            return self._kokkos_state.ExpectationValue(
+                CSR_SparseHamiltonian.data,
+                CSR_SparseHamiltonian.indices,
+                CSR_SparseHamiltonian.indptr,
+            )
+
+        if observable.name in ["Hamiltonian"] and len(observable.wires) < 13:
+            device_wires = self.map_wires(observable.wires)
+            return self._kokkos_state.ExpectationValue(
+                device_wires, qml.matrix(observable).ravel(order="C")
+            )
```

Comment:
We also still need the case for wires >=13: it should fall-back to convert the dense H to sparse and use the earlier sparse method.

```
Hmat = qml.utils.sparse_hamiltonian(observable, wires=self.wires)
```

Should do it.

Reply:
I cannot see the implementation for this. Was it resolved and put elsewhere?

Reply:
Oh weird, forgot to push